### DISCUSSION TITLE
WIkiJS Fails to Update Discord Callback URI

### DISCUSSION THREAD

**Discussion by invictus2010** [requarks/wiki#5278] [Error / Bug Report] [answered]
Appears that WikiJS is generating or redirecting to the incorrect callback URI for Discord auth strategy. On WikiJS version 2.5.281.

## Steps to Reproduce
1. Follow steps A-C of [guide](https://docs.requarks.io/auth/discord) to enabling Discord auth strategy. Note the callback URI copied from WikiJS admin pane to the Discord dev panel.
2. Attempt to login with Discord.
3. Receive error `Invalid OAuth2 redirect_uri`
4. Examine URL. URL is `https://discord.com/oauth2....wiki.yourdomain.com...`
5. Verified that URL in Discord auth strategy in WikiJS from Step C of guide does not contain **yourdomain.com**

## Temporary Workaround
1. Go to the URL bar
2. Find the string...wiki.yourdomain.com
3. Replace yourdomain with correct text.
4. Refresh/hit enter
5. Continue with login

**Comment by jarrodlovas** [comment]
When you add the discord auth, at the bottom, did you get the callback URL?
![image](https://user-images.githubusercontent.com/47882472/168056617-c528fdb4-7c4b-42fd-beac-1fc547b04825.png)

**Comment by invictus2010** [comment]
<img width="828" alt="Screen Shot 2022-05-12 at 9 33 25 AM" src="https://user-images.githubusercontent.com/17661001/168087356-d0120957-ab0f-453e-8a1b-32c47cddb434.png">

I then put this URL into my Discord developer panel here.

<img width="773" alt="Screen Shot 2022-05-12 at 9 35 27 AM" src="https://user-images.githubusercontent.com/17661001/168087693-95dc3cef-3008-420c-8b51-56cb5a256593.png">

Nevertheless, when a user attempts to use Discord auth to login, the redirect URL is wiki.yourdomain.com, not wiki.<actualUrl>.com

**Comment by invictus2010** [comment, marked as answer]
Resolved the issue by performing a reboot of the server.

Likely needed to perform reboot after updating the website URL field found in the general settings.
